internal/pkg/skuba/addons/cilium.go

```diff
@@ -74,7 +95,24 @@ func (ciliumCallbacks) beforeApply(addonConfiguration AddonConfiguration, skubaC
 			return err
 		}
 	}
-	if err := cni.CreateOrUpdateCiliumConfigMap(client); err != nil {
+
+	// Check whether data migration from etcd to CRD is needed and if yes,
+	// perform it.
+	migrationNeeded, err := cni.IsMigrationToCrdNeeded(client)
```

Comment:
I wonder if we could group all actions that must happen when doing an upgrade in a more general way, so that when we need to do the upgrade of 1.6 --> 1.7, we can reuse part of the code. I was thinking on something like this:
```
if upgradeRequired {
   if (currentCiliumVersion == 1.5 && targetedCiliumVersion == 1.6) {
      cni.MigrateEtcdToCrd(client, config, manifest)
      otherAction()
   }
   if (currentCiliumVersion == 1.6 && targetedCiliumVersion == 1.7) {
       upgradeAction6To7()
   } 
}
```